runtime/src/iree/builtins/device/arch/arm_64/sme_abi.c

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "../../device.h"
+
+// This file contains stubs of the AArch64 SME ABI support routines, defined in
+// the AAPCS64.
+// See:
+// https://github.com/ARM-software/abi-aa/blob/main/aapcs64/aapcs64.rst#81sme-support-routines.
+//
+// Note that use use of these stubs is safe for the code IREE currently
+// generates, as each dispatch region is self-contained. SME dispatches do not
+// call out to other SME-enabled functions, so ZA-state does not need to be
+// preserved.
+//
+// The LLVM backend still plants calls to these functions, so these definitions
+// are simply to satisfy the linker, but these functions won't ever be called.
+
+typedef struct sme_state {
+  int64_t x0;
+  int64_t x1;
+} sme_state_t;
+
+typedef struct TPIDR2_block TPIDR2_block_t;
+
+IREE_DEVICE_EXPORT sme_state_t __arm_sme_state(void) {
+  // TODO(macdue): Implement.
```

Comment:
Given we don't expect SME-enabled dispatch functions with live ZA state to call other such functions, I think we can remove the TODO's, as these stubs are sufficient?

Reply:
I think it's still a TODO, even if we don't need these functions (yet), that might not always be the case.

Reply:
with TODOs we try to make them actionable (in case the originating author disappears) - a comment of "// No-op today as not needed." would be informative/durable and let people know this file is good as-is, while the TODO without context makes it look like it's not done yet.